context-free.md

```diff
@@ -1,22 +1,30 @@
 # Context Free
 
 ##  Who is this programming language for?
-
+It is for image designers or graphical artists. 
 
 ## What is easy to do in this language? Why is it easy?
-
+It is really easy to make squares, circles, and triangles of various colors, shapes, sizes, and locations.
+Its also very easy to make other shapes that build off triangles, circles, or squares to make a 
+more complicated figure. 
 
 ## What is hard to do in this language? Why is it hard?
-
+Firstly, its impossible to do anything but make images (or so it seems) . Secondly, I thought it was hard to make non circular, square, or triangular shapes at because the base shapes are very basic. For example, to make a smile I had to use miniature squares put together which is not very intuitive. 
 
 ## How did you learn how to program in this language?
-
+The documentation confused me so I mainly looked at examples to figure out how to change the colors and the different syntax.
 
 ## What is the underlying _computational model_ for this programming language? 
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of what 
 happens when a ContextFree program runs._
 
+WHen I hit render, a text parser figures out the rule definitions written and 
+what rules they are based off of (for example my arc is based off of squares). 
+It then probably uses the rule definitions to determine which pixels need to be colored. 
+Then using the coloring definitions, it puts a color at that pixel in the canvas. 
 
-## What do you think is interesting about the ContextFree program you wrote?
 
+## What do you think is interesting about the ContextFree program you wrote?
```

Comment:
It is certainly interesting how you used little squares to make the smiley face. In the documentation, there is a section about [paths](https://github.com/MtnViewJohn/context-free/wiki/Path-Declarations) in which you can make lines and other stuff. I haven't looked to much into it, and it might not be available in version 2.2. However, this is my guess in how they create curves such as the recursive flower example.